hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -630,8 +642,30 @@ def generate_new_examples(self):
 
         self.health_check_state = HealthCheckState()
 
+        def should_generate_more():
+            assert self.first_bug_calls <= self.last_bug_calls <= self.call_count
+            # If we haven't found a bug yet, keep going until the loop is broken by
+            # `self.exit_with(ExitReason.max_examples)` in `self.test_case`.
+            if not self.interesting_examples:
+                return True
+            if (
+                # Never try to generate more than max_examples examples, or
+                # generate disallowed shrinks in the generation phase.
+                self.valid_examples >= self.settings.max_examples
+                or Phase.shrink not in self.settings.phases
```

Comment:
I'm a bit weirded out by this use of `Phase.shrink`. I don't think reusing it to include secondary generation makes a lot of sense.

Reply:
Better to add a new phase then? Or just bite the bullet and hunt down all the places we assume disabling shrinking caps the number of test cases?